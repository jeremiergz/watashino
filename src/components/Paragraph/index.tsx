import Text, { TextProps } from 'components/primitives/Text';
import React from 'react';

const Paragraph: React.FC<ParagraphProps> = ({ children, ...rest }) => {
  return (
    <Text as="p" fontSize={20} marginY={4} textAlign={{ _: 'center', mobileL: 'justify' }} {...rest}>
      {children}
    </Text>
  );
};

Paragraph.displayName = 'Paragraph';

export type ParagraphProps = TextProps;
export default Paragraph;
