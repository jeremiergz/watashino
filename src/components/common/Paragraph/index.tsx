import React from 'react';
import Text, { TextProps } from '../../primitives/Text';

type ParagraphProps = TextProps;

const Paragraph = ({ children, ...rest }: ParagraphProps) => {
  return (
    <Text as="p" fontSize={20} marginY={4} textAlign={{ _: 'center', mobileL: 'left' }} {...rest}>
      {children}
    </Text>
  );
};

Paragraph.displayName = 'Paragraph';

export { ParagraphProps };
export default Paragraph;
