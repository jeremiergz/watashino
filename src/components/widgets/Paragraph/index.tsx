import React, { HTMLAttributes } from 'react';
import Text from '../../primitives/Text';

type Props = HTMLAttributes<HTMLParagraphElement> & {};

const ParagraphComponent = ({ children, ...rest }: Props) => {
    return (
        <Text as="p" fontSize={20} textAlign="left" {...rest}>
            {children}
        </Text>
    );
};

ParagraphComponent.displayName = 'Paragraph';

export default ParagraphComponent;
