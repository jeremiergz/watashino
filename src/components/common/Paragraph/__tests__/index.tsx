import React from 'react';
import ReactTestRenderer from 'react-test-renderer';
import Paragraph from '../../Paragraph';

describe('components/common/Paragraph Test Suite', () => {
  it('matches snapshot', () => {
    const jsx = <Paragraph>Paragraph Component</Paragraph>;
    expect(ReactTestRenderer.create(jsx).toJSON()).toMatchSnapshot();
  });
});
