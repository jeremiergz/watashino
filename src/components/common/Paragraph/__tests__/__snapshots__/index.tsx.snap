// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/widgets/Paragraph Test Suite matches snapshot 1`] = `
<p
  className="sc-AxjAm hPZwzt"
  fontSize={20}
>
  Paragraph Component
</p>
`;
